packages/wallet-sdk/src/relay/aes256gcm.test.ts

```diff
@@ -31,11 +32,12 @@ describe("aes256gcm", () => {
       },
     };
 
-    decrypt(cipherText, secret).subscribe({
-      next: value => {
-        expect(sampleDataResult).toEqual(value);
+    decrypt(cipherText, secret).then(
+      value => {
+        expect(sampleDataResult).toEqual(JSON.parse(value));
```

Comment:
wonder how this has been working
decrypt returns Promise (or used to be Observable) of string, so `value` (string) and `sampleDataResult` (object) couldn't be equal, right?
![image](https://user-images.githubusercontent.com/966662/211397944-b538af3a-e4d1-4253-aadc-34113d9f77ee.png)

Reply:
did the observable resolve to the string or was it an object?

Reply:
I just logged out what the old test returns when run and it seems like it actually returns an object. Incorrect typing on the decrypt function?

<img width="1047" alt="image" src="https://user-images.githubusercontent.com/3137711/211409559-77f3f28a-a0ce-4109-8425-8d89d57a8530.png">

Reply:
@vishnumad i am getting a string with old code
<img width="845" alt="image" src="https://user-images.githubusercontent.com/966662/211417988-a45c8339-38ad-4cff-8d77-b646a4c63044.png">
<img width="647" alt="image" src="https://user-images.githubusercontent.com/966662/211418098-82ac11b2-45dd-41ae-983b-889eeae8ef12.png">

Reply:
@bangtoven Yup, you're correct it is a string. This test is just broken and will pass no matter what you do in the subscribe function because it's an observable. You can test this by replacing the current expect with `expect(true).toEqual(false);`. We need to actually block until the observable emits the value.

```js
 test("decrypt", done => {
    const cipherText =
      "06593325a922a928913b5c6ea26f848c4545bcea4e26c4f5ee745316ff22b2780aeccc565730514b2820a94b03f5f89fe7542a35bbdd87a1d52a4352f49482781113db09266c668696778e0a94bc9f866f1e92e7262fd0bb811838284cc64cbc4552b33e9c6fb2582cea4f49471d6d46a16a5c8ac83ee8483ed4dc01f1fde3bfd7a2f173715e0a8d09dd4907483f096a845bff698831ea277c1ca4223d3f6073174cb35119d0a795c1a9cb4f32ee1dcc254d8931";
    const sampleDataResult = {
      type: "WEB3_RESPONSE",
      id: "791fe0ec3dc3de49",
      response: {
        method: "requestEthereumAccounts",
        result: ["0xdf0635793e91d4f8e7426dbd9ed08471186f428d"],
      },
    };

    decrypt(cipherText, secret).subscribe({
      next: value => {
        expect(sampleDataResult).toEqual(value);
        done();
      },
    });
  });
```

Reply:
@bangtoven You'll run into a similar issue with the promises in the current test. The test won't actually get marked as failed currently since you're not awaiting the result of the promise. 
<img width="1178" alt="image" src="https://user-images.githubusercontent.com/3137711/211421754-0514806f-8a7d-463b-afc7-4e8229e3db3f.png">

```js
test("decrypt", async () => {
    // ...
    const value = await decrypt(cipherText, secret);
    expect(value).toEqual(sampleDataResult);
  });
```

Reply:
@vishnumad hmm. actually i was able to find type error since the test failed.

Reply:
@bangtoven It does print the error to console, but doesn't mark the test as failed. See the orange `RUNS` next to the last line in the screenshot above vs red `FAIL` below when using await
<img width="1173" alt="image" src="https://user-images.githubusercontent.com/3137711/211423180-0e78ac93-6c1f-46ee-8b61-fa7d10c38a29.png">

Reply:
oh, gotcha. i just made change to the test code to return the promise.then block.